torchgeo/datasets/geo.py

```diff
@@ -525,34 +532,89 @@ def __getitem__(self, query: GeoSlice) -> dict[str, Any]:
+                            if match:
+                                if 'band' in match.groupdict():
+                                    start = match.start('band')
+                                    end = match.end('band')
+                                    filename = filename[:start] + band + filename[end:]
+                            filepath = os.path.join(directory, filename)
+                            band_filepaths.append(filepath)
+
+                        # Merge files for this band at this time step
+                        band_data.append(self._merge_files(band_filepaths, query))
+
+                    # Concatenate all bands for this time step
+                    time_step_data = torch.cat(band_data, dim=0)
+                else:
+                    # For non-separate files, merge all files for this time step
+                    time_step_data = self._merge_files(
+                        group.filepath, query, self.band_indexes
+                    )
+
+                time_steps.append(time_step_data)
+
+            # Stack along time dimension to create [T,C,H,W]
+            data = torch.stack(time_steps, dim=0)
+
         else:
-            data = self._merge_files(index.filepath, query, self.band_indexes)
+            # Original non-timeseries behavior
+            if self.separate_files:
+                data_list: list[Tensor] = []
+                filename_regex = re.compile(self.filename_regex, re.VERBOSE)
+                for band in self.bands:
+                    band_filepaths = []
+                    for filepath in index.filepath:
+                        filename = os.path.basename(filepath)
+                        directory = os.path.dirname(filepath)
+                        match = re.match(filename_regex, filename)
+                        if match:
+                            if 'band' in match.groupdict():
+                                start = match.start('band')
+                                end = match.end('band')
+                                filename = filename[:start] + band + filename[end:]
+                        filepath = os.path.join(directory, filename)
+                        band_filepaths.append(filepath)
+                    data_list.append(self._merge_files(band_filepaths, query))
+                data = torch.cat(data_list)
+            else:
+                data = self._merge_files(index.filepath, query, self.band_indexes)
 
         sample: dict[str, Any] = {'crs': self.crs, 'bounds': query}
+        if self.time_series:
+            sample['dates'] = dates
 
         data = data.to(self.dtype)
         if self.is_image:
             sample['image'] = data
         else:
-            sample['mask'] = data.squeeze(0)
+            # Try to squeeze singleband mask
+            if self.time_series:
+                sample['mask'] = data.squeeze(1)
+            else:
+                sample['mask'] = data.squeeze(0)
```

Comment:
@ashnair1 do you remember why we added this? In general, I don't like using [un]squeeze, as it can have unintended consequences.

Reply:
My guess was to drop the single channel dimension, but not sure why that was initially in there, I just followed what was there.